pygen/pygen_src/riscv_privileged_common_seq.py

```diff
@@ -102,20 +105,82 @@ def setup_mmode_reg(self, mode, regs):
             self.mie.set_field("UTIE", cfg.enable_interrupt & cfg.enable_timer_irq)
             regs.append(self.mie)
 
-    # TODO
     def setup_smode_reg(self, mode, regs):
-        pass
+        self.sstatus = riscv_privil_reg()
+        self.sstatus.init_reg(privileged_reg_t.SSTATUS)
+        self.sstatus.randomize()
+        if cfg.randomize_csr:
+            self.sstatus.set_val(cfg.sstatus)
+        self.sstatus.set_field("SPIE", cfg.enable_interrupt)
+        self.sstatus.set_field("SIE", cfg.enable_interrupt)
+        self.sstatus.set_field("UPIE", cfg.enable_interrupt)
+        self.sstatus.set_field("UIE", rcs.support_umode_trap)
+        if rcs.XLEN == 64:
+            self.sstatus.set_field("UXL", 2)
+        self.sstatus.set_field("FS", cfg.mstatus_fs)
+        self.sstatus.set_field("XS", 0)
+        self.sstatus.set_field("SD", 0)
+        self.sstatus.set_field("UIE", 0)
+        self.sstatus.set_field("SPP", 0)
+        regs.append(self.sstatus)
+        # Enable external and timer interrupt
+        if "SIE" in rcs.implemeted_csr:
+            self.sie = riscv_privil_reg()
+            self.sie.init_reg(privileged_reg_t.SIE)
+            if cfg.randomize_csr:
+                self.sie.set_val(cfg.sie)
+            self.sie.set_field("UEIE", cfg.enable_interrupt)
+            self.sie.set_field("SEIE", cfg.enable_interrupt)
+            self.sie.set_field("USIE", cfg.enable_interrupt)
+            self.sie.set_field("SSIE", cfg.enable_interrupt)
+            self.sie.set_field("STIE", cfg.enable_interrupt & cfg.enable_timer_irq)
+            self.sie.set_field("UTIE", cfg.enable_interrupt & cfg.enable_timer_irq)
```

Comment:
For better readability, perhaps better to create a separate method to do all the `set_fields` (for `self.sstatus` as well as `self.sie`)

Reply:
Hi @hodjat91 
These are itself in a method which is being called from the `enter_privileged_mode` and using `status` and `sie` objects I am calling the `set_fields` method.
Even If I create new methods for these, The line of code and other things would be the same.
What do you think?

Reply:
Hi @hodjat91 
I have changed this as you suggested.
Please do have a look!